modules/tutorials/pages/java-tutorial/adding-enrollments.adoc

```diff
@@ -0,0 +1,202 @@
+
+        enrollments.add(JsonObject.create().put("course-id", art_history.getString("id"))
+                .put("date-enrolled", currentDate));    // <8>
+
+        hilary.put("enrollments", enrollments);     // <9>
+
+        student_records.upsert(hilary.getString("id"), hilary);    // <10>
+
+        cluster.disconnect();
+
+    }
+
+    private static JsonObject retrieveStudent(Cluster cluster, String name) throws CouchbaseException {
+
+        final QueryResult result = cluster.query("select META().id, src.* " +
+                        "from `student-bucket`.`art-school-scope`.`student-record-collection` src " +
+                        "where src.`name` = $name",
+                QueryOptions.queryOptions()
+                        .parameters(JsonObject.create().put("name", name)));
+
+        return result.rowsAsObject().get(0);
+
+    }
+
+    private static JsonObject retrieveCourse(Cluster cluster, String course) throws CouchbaseException {
+
+        final QueryResult result = cluster.query("select META().id, crc.* " +
+                "from `student-bucket`.`art-school-scope`.`course-record-collection` crc " +
+                "where crc.`course-name` = $courseName",
+                QueryOptions.queryOptions()
+                        .parameters(JsonObject.create().put("courseName", course)));
+
+        return result.rowsAsObject().get(0);
+
+    }
+
+}
+
+----
+<1> As you'll remember from our xref:java-tutorial/first-couchbase-application.adoc#connecting-to-the-database[first example], the application first has to connect to the Couchbase cluster
+
+<2> Then it picks up the correct `Bucket` from the cluster.
+
+<3> Do you remember why we need this `waitUntilReady` function is needed when connecting to the server?
++
+{empty}
++
+[%collapsible]
+.Answer
+====
+{why-waitUntilReady}
+====
++
+{empty}
+<4> Here again, we pick up the student collection from the `art-school-scope`; you'll need it later to change Hilary's record and write it back to the collection.
+
+<5> There are three method calls here:
++
+--
+. Retrieve Hilary's student record
+. Retrieve the `graphic design` course record
+. Retrieve the `art history` course record
+--
++
+Each method uses a `N1QL` call to retrieve a single record from its respective collection.
++
+IMPORTANT: This is just a demonstration, so there is no error checking to ensure that an item from the collection has been returned. In a live system, checks would have to be made to prevent possible errors while the program is running.
++
+{empty}
+<6> Remember that Couchbase doesn't have a native date type, so it's common practice to store the dates as strings.
+
+<7> The `enrollments` inside the student record are stored as an array; since `Json` is the native data format for Couchbase, it's not surprising that each SDK has a host of functions for processing data in Json. In this case, the call `JsonArray.create()` will create an empty list structure.
```

Comment:
When referring to the data format, JSON in capitals, doesn't need to be formatted as code.